nacc_form_validator/nacc_validator.py

```diff
@@ -690,62 +690,58 @@ def _validate_function(self, function: str, field: str, value: object):
             self.__add_system_error(field, err_msg)
             raise ValidationException(err_msg)
 
-    def _check_with_gds(self, field: str, value: object):
+    def _validate_compute_gds(self, keys: List[str], field: str, value: object):
         """Validate Geriatric Depression Scale (GDS) calculation.
 
         Args:
+            keys: GDS computation fields specified in the rule definition
             field: Variable name
             value: Variable value
+
+        Note: Don't remove below docstring,
+        Cerberus uses it to validate the schema definition.
+
+        The rule's arguments are validated against this schema:
+            {
+                'type': 'list', 
+                'minlength': 15,
+                'schema': {'type': 'string'}
+            }
         """
 
         nogds = 0
         if "nogds" in self.document:
             nogds = self.document["nogds"]
 
-        if nogds == 1 and value != 88:
-            self._error(field, ErrorDefs.CHECK_GDS_1)
-            return
-
-        keys = [
-            "satis",
-            "dropact",
-            "empty",
-            "bored",
-            "spirits",
-            "afraid",
-            "happy",
-            "helpless",
-            "stayhome",
-            "memprob",
-            "wondrful",
-            "wrthless",
-            "energy",
-            "hopeless",
-            "better",
-        ]
-
         num_valid = 0
         gds = 0
         for key in keys:
             if key in self.document and self.document[key] in [1, 0]:
                 num_valid += 1
                 gds += self.document[key]
 
-        if nogds == 1 and num_valid >= 12:
-            self._error(field, ErrorDefs.CHECK_GDS_2)
+        if nogds == 1:
+            if num_valid >= 12:
+                self._error(field, ErrorDefs.CHECK_GDS_1, 0)
+
+            if value != 88:
```

Comment:
Can we reorder the error/logic checks here to be the same order as the CSV/codes so that they're also in order when we map the codes: https://github.com/naccdata/uniform-data-set/blob/3ad751fc46a3a8590ecf68d7ab0d0d3357829167/forms/uds/b6/form_b6_ivp_error_checks_p.csv

Most notably 1 and 2 need to be swapped. Might also want to ask RT to move 1005 (if GDS is blank there must be at least 12 questions with valid scores) to be 1003 instead.